src/laboratory/api/views.py

```diff
@@ -702,3 +713,200 @@ def list(self, request, *args, **kwargs):
+
+        # Multiple response data
+        response_data = material_serializer.data
+        material_ca_data = material_ca_serializer.data
+
+        # THIS ID SHOULDN'T REPLACE THE MAIN ID(EQUIPMENT OBJECT)
+        del material_ca_data['id']
+        response_data.update(material_ca_data)
+
+        return response_data, material_changed_data, material_ca_changed_data
+
+    def create(self, request, *args, **kwargs):
+        self.org_pk = kwargs["org_pk"]
+        self.lab_pk = kwargs["lab_pk"]
+        organization = get_object_or_404(
+            OrganizationStructure.objects.using(settings.READONLY_DATABASE),
+            pk=self.org_pk)
+        errors, response_data, headers = {}, {}, self.get_success_headers({})
+
+        # Serializers
+        material_serializer = self.get_serializer(data=request.data)
+        material_ca_serializer = ValidateMaterialCapacitySerializer(
+            data=request.data, context={"lab_pk": self.lab_pk})
+
+        if material_serializer.is_valid():
+            if material_ca_serializer.is_valid():
+                instance = material_serializer.save()
+                material_ca_serializer.save(object=instance)
+
+                response_data, material_changed_data, material_ca_changed_data = self.get_response_validate_data(
+                    material_serializer, material_ca_serializer)
+
+                # Multiple headers
+                headers = self.get_success_headers(response_data)
+
+                # Log Entry Create Action
+                organilab_logentry(request.user, instance, ADDITION, "material object",
+                                   changed_data=material_changed_data,
+                                   relobj=organization)
+
+                if hasattr(instance, 'materialcapacity'):
+                    organilab_logentry(request.user, instance, ADDITION,
+                                       "material capacity",
+                                       changed_data=material_ca_changed_data,
+                                       relobj=organization)
+
+                return Response(response_data, status=status.HTTP_201_CREATED,
+                                headers=headers)
+            else:
+                errors.update(material_ca_serializer.errors)
+        else:
+            errors.update(material_serializer.errors)
+            if not material_ca_serializer.is_valid():
+                errors.update(material_ca_serializer.errors)
+
+        if errors:
+            raise ValidationError(errors)
+
+    def get_material_ca_serializer(self, instance, request, partial, lab_pk):
+        data = request.data
+
+        if hasattr(instance, 'materialcapacity'):
+            data['object'] = instance.pk
+            material_ca_instance = instance.materialcapacity
+            material_ca_serializer = ValidateMaterialCapacitySerializer(
+                material_ca_instance, data=data, partial=partial,
+                context={"lab_pk": lab_pk})
```

Comment:
Why you are sending lab_pk by context if in this serializer is not using this parameter.